explainability-core/src/main/java/org/kie/trustyai/explainability/model/DataframeMetadata.java

```diff
@@ -0,0 +1,193 @@
+package org.kie.trustyai.explainability.model;
+
+import org.kie.trustyai.explainability.model.domain.EmptyFeatureDomain;
+import org.kie.trustyai.explainability.model.domain.FeatureDomain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class DataframeMetadata {
+    private final List<String> names;
+    private List<String> nameAliases;
+    private List<String> cachedColumnNames;
+    private List<Type> types;
+    private final List<Boolean> constrained;
+    private List<FeatureDomain> domains;
+    private final List<Boolean> inputs;
+
+    // constructors ====================================================================================================
+    public DataframeMetadata() {
+        this(new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
+    }
+
+    public DataframeMetadata(List<String> names, List<String> nameAliases, List<Type> types, List<Boolean> constrained, List<FeatureDomain> domains,
```

Comment:
I would probably add some null checking here. Empy lists should be supported, but it should throw an error on null values.

Reply:
done